### PR TITLE
Fix/ET-1821 do not allow free tickets commerce

### PR DESCRIPTION
**Pull request by estebanhdzc**
### 🎫 Ticket

[ET-1821] 

### 🗒️ Description
Bug fix: 
**Backend**
- Added a validation in `[POST]` `/tribe/tickets/v1/tickets/` endpoint to prevent create tickets with price equals or lower than 0  when provider is `Tribe__Tickets__Commerce__PayPal__Main` or `TEC\Tickets\Commerce\Module`

**Frontend**
- Added TEC\Tickets\Commerce\Module as constant
- Updated isZeroPriceValid selector
- Modified price input min value depending on provider

### 🎥 Artifacts 
https://www.loom.com/share/de7b944851c8435e8831f6e3c1256bbe

### ✔️ Checklist
- [x] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1821]: https://theeventscalendar.atlassian.net/browse/ET-1821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/data/blocks/ticket/constants.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const WOO = 'woo';
 export const RSVP = 'rsvp';
 
 export const RSVP_CLASS = 'Tribe__Tickets__RSVP';
+export const TEC_TICKETS_COMMERCE_MODULE_CLASS = 'TEC\\Tickets\\Commerce\\Module';
```
Comment (bordoni) [inline src/modules/data/blocks/ticket/constants.js:7]: I would avoid `TEC_` prefix on this case.

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php [hunk 2]
```diff
@@ -254,6 +255,17 @@ public function add_ticket( WP_REST_Request $request, $nonce_action ) {
 			);
 		}
 
+		$is_paypal_ticket = $provider instanceof Tribe__Tickets__Commerce__PayPal__Main || $provider instanceof \TEC\Tickets\Commerce\Module;
+		$price            = (float)$body['price'];
+
+		if ( $is_paypal_ticket && $price <= 0) {
+			return new WP_Error(
+				'bad_request',
```
Comment (bordoni) [inline src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php:263]: Maybe we should use our own error slug.

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php [hunk 3]
```diff
@@ -254,6 +255,17 @@ public function add_ticket( WP_REST_Request $request, $nonce_action ) {
 			);
 		}
 
+		$is_paypal_ticket = $provider instanceof Tribe__Tickets__Commerce__PayPal__Main || $provider instanceof \TEC\Tickets\Commerce\Module;
+		$price            = (float)$body['price'];
```
Comment (bordoni) [inline src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php:259]: Avoid converting prices to Float (calculation errors), even if here is just to make a check it could cause problems in the future if someone uses this variable.

Do a couple checks, like `is_numeric` and something like using the float casting should be used directly on conditional so we dont create a new variable that could be misused.

File: tests/restv1/PayPal/SingleTicketCest.php [hunk 4]
```diff
@@ -405,4 +405,128 @@ public function should_return_401_when_trying_to_access_non_public_ticket( Restv
 		$I->seeResponseCodeIs( 401 );
 		$I->seeResponseIsJson();
 	}
+
+	/**
+	 * It should return 400 when trying to create a ticket with invalid price
+	 *
+	 * @test
+	 */
+	public function should_return_400_when_trying_to_create_a_ticket_ticket_with_invalid_price( Restv1Tester $I ) {
```
Comment (bordoni) [inline tests/restv1/PayPal/SingleTicketCest.php:414]: I like those tests, I would also love to see us test creation with variations of Zero.

Like empty string, "", `0.0` , `0,0`, `000` and ` `

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/modules/data/blocks/ticket/constants.js (hunk 1 above)
Comment (estebanhdzc) [reply src/modules/data/blocks/ticket/constants.js]: Thanks! Constant was renamed

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php (hunk 3 above)
Comment (estebanhdzc) [reply src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php]: Creation of the variable was avoided. Instead I did the validation against `$body['price']` directly

File: tests/restv1/PayPal/SingleTicketCest.php (hunk 4 above)
Comment (estebanhdzc) [reply tests/restv1/PayPal/SingleTicketCest.php:414]: More tests were added

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php (hunk 2 above)
Comment (estebanhdzc) [reply src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php]: Slug was changed

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php [hunk 5]
```diff
@@ -254,6 +255,16 @@ public function add_ticket( WP_REST_Request $request, $nonce_action ) {
 			);
 		}
 
+		$is_paypal_ticket = $provider instanceof Tribe__Tickets__Commerce__PayPal__Main || $provider instanceof \TEC\Tickets\Commerce\Module;
+
+		if ( $is_paypal_ticket && ( ! is_numeric( $body['price'] ) || (float)$body['price'] <= 0 )) {
```
Comment (bordoni) [inline src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php:260]:
```suggestion
		$is_invalid_price = ( empty( $body['price']  ) || ! is_numeric( $body['price'] ) || (float) $body['price'] <= 0 );
		if (
			$is_paypal_ticket 
			&& $is_invalid_price
		) {
```

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php (hunk 5 above)
Comment (estebanhdzc) [reply src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php]: Suggestion applied. Thanks for noticing this one!

**Review by bordoni — CHANGES_REQUESTED** [review]
One last thing.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php [hunk 6]
```diff
@@ -254,6 +255,20 @@ public function add_ticket( WP_REST_Request $request, $nonce_action ) {
...
 
+		$is_paypal_ticket = $provider instanceof Tribe__Tickets__Commerce__PayPal__Main || $provider instanceof \TEC\Tickets\Commerce\Module;
+		$is_invalid_price = ( empty( $body['price']  ) || ! is_numeric( $body['price'] ) || (float) $body['price'] <= 0 );
+
+		if (
+			$is_paypal_ticket
+			&& $is_invalid_price
+		) {
+			return new WP_Error(
+				'invalid_price',
```
[the remark this answers is not in the document]
Comment (estebanhdzc) [reply src/Tribe/Editor/REST/V1/Endpoints/Single_Ticket.php]: Thanks! This is fixed now

**Review by bordoni — APPROVED** [review]
